binding/ios/PorcupineManager.swift

```diff
@@ -11,150 +11,234 @@
-    init(modelFilePath: String, keywordFilePath: String, sensitivity: Float, keywordCallback: @escaping (() -> Void)) {
+    ///   - wakeKeywordConfigurations: Keyword configurations to use.
+    ///   - onDetection: Detection handler to call after wake word detection. The handler is executed on main thread.
+    /// - Throws: PorcupineManagerError
+    public init(modelFilePath: String, wakeKeywordConfigurations: [WakeKeywordConfiguration], onDetection: ((WakeKeywordConfiguration) -> Void)?) throws {
+
         self.modelFilePath = modelFilePath
-        self.keywordFilePaths = [keywordFilePath]
-        self.sensitivities = [sensitivity]
-        self.keywordCallback = keywordCallback
-        self.multiKeywordCallback = nil
+        self.wakeKeywordConfigurations = wakeKeywordConfigurations
+        self.onDetection = onDetection
+
+        let keywordFilePaths = wakeKeywordConfigurations.map { $0.filePath }
+        let sensitivities = wakeKeywordConfigurations.map { $0.sensitivity }
+
+        let status = pv_porcupine_multiple_keywords_init(
+            modelFilePath,
+            Int32(keywordFilePaths.count),
+            keywordFilePaths.map { UnsafePointer(strdup($0)) },
+            sensitivities,
+            &porcupine)
+        try checkInitStatus(status)
+
+        let audioSession = AVAudioSession.sharedInstance()
+
+        NotificationCenter.default.addObserver(self, selector: #selector(onAudioSessionInterruption(_:)), name: .AVAudioSessionInterruption, object: audioSession)
     }
 
-    
-    /// Initializer for multiple keyword use case.
+    /// Initializer for single keyword detection.
     ///
     /// - Parameters:
     ///   - modelFilePath: Absolute path to file containing model parameters.
-    ///   - keywordFilePaths: Absolute paths to keyword files.
-    ///   - sensitivities: List of sensitivity parameters for each keyword.
-    ///   - keywordCallback: Callback to be executed after wake word detection.
-    init(modelFilePath: String, keywordFilePaths: [String], sensitivities: [Float], keywordCallback: @escaping ((Int32) -> Void)) {
-        self.modelFilePath = modelFilePath
-        self.keywordFilePaths = keywordFilePaths
-        self.sensitivities = sensitivities
-        self.multiKeywordCallback = keywordCallback
-        self.keywordCallback = nil
+    ///   - wakeKeywordConfigurations: Keyword configuration to use.
+    ///   - onDetection: Detection handler to call after wake word detection. The handler is executed on main thread.
+    /// - Throws: PorcupineManagerError
+    public convenience init(modelFilePath: String, wakeKeywordConfiguration: WakeKeywordConfiguration, onDetection: ((WakeKeywordConfiguration) -> Void)?) throws {
+        try self.init(modelFilePath: modelFilePath, wakeKeywordConfigurations: [wakeKeywordConfiguration], onDetection: onDetection)
+    }
+
+    deinit {
+        pv_porcupine_delete(porcupine)
```

Comment:
I wonder if this should be done the other way around. First, stop listening and then delete the porcupine object. the reason is that there is a chance that the object is called after being deleted that can cause problems. I might be missing something here ...